[ews] Convert CompileWebKit to new style buildstep

aj062 opened the pull request:
#### f30213fb0e5b1aacfe582a98cf134eef862d81b6
<pre>
[ews] Convert CompileWebKit to new style buildstep
<a href="https://bugs.webkit.org/show_bug.cgi?id=298250">https://bugs.webkit.org/show_bug.cgi?id=298250</a>
<a href="https://rdar.apple.com/159683955">rdar://159683955</a>

Reviewed by Brianna Fan.

Similar to <a href="https://commits.webkit.org/298343@main">https://commits.webkit.org/298343@main</a>
Old-style steps are officially deprecated and are removed in newer versions of buildbot.

* Tools/CISupport/ews-build/steps.py:
(CompileWebKit):
(CompileWebKit.run):
(CompileWebKit.errorReceived): Similar to <a href="https://commits.webkit.org/298526@main">https://commits.webkit.org/298526@main</a>
(CompileWebKit.getResultSummary):
(CompileWebKitWithoutChange.evaluateCommand): Ensure this class doesn&apos;t call evaluateCommand from CompileWebKit.
Explicitly called it from shell.CompileNewStyle.
(CompileJSC.getResultSummary): Ditto.
(CompileJSC32.getResultSummary): Ditto.
(CompileJSCWithoutChange.evaluateCommand): Ditto.
(CompileJSCWithoutChange32.evaluateCommand): Ditto.

Canonical link: <a href="https://commits.webkit.org/299458@main">https://commits.webkit.org/299458@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b7041aad63e6f9e799c614f326281f3717614dfb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119052 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38733 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29384 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/125264 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71115 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7def3b0b-7208-4c37-ae2b-09d74f7ff92e) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120930 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39429 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/47315 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90395 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59847 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c76b62fb-7434-47a0-921c-66300e456df0) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122005 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31433 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106730 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/70855 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/029e0a58-9001-4ecc-8b51-3be4c7938281) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30495 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24846 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68915 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100880 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25032 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/128293 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45959 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34731 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99017 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/118476 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/46326 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102950 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98796 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44251 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22253 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/42538 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18958 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45829 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/45296 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48641 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46981 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7277251272a7b0b7d1dac9fbef630f5a52697a73. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50188

aj062 committed:
[ews] Convert CompileWebKit to new style buildstep
https://bugs.webkit.org/show_bug.cgi?id=298250
rdar://159683955

Reviewed by Brianna Fan.

Similar to https://commits.webkit.org/298343@main
Old-style steps are officially deprecated and are removed in newer versions of buildbot.

* Tools/CISupport/ews-build/steps.py:
(CompileWebKit):
(CompileWebKit.run):
(CompileWebKit.errorReceived): Similar to https://commits.webkit.org/298526@main
(CompileWebKit.getResultSummary):
(CompileWebKitWithoutChange.evaluateCommand): Ensure this class doesn't call evaluateCommand from CompileWebKit.
Explicitly called it from shell.CompileNewStyle.
(CompileJSC.getResultSummary): Ditto.
(CompileJSC32.getResultSummary): Ditto.
(CompileJSCWithoutChange.evaluateCommand): Ditto.
(CompileJSCWithoutChange32.evaluateCommand): Ditto.

Canonical link: https://commits.webkit.org/299458@main
